### PR TITLE
[python] Fix and test over-indexing semantics

### PR DESCRIPTION
**Pull request by johnkerl**
## Issue and/or context:

#900

### CODE REVIEW

File: apis/python/src/tiledbsoma/util.py [hunk 1]
```diff
@@ -128,6 +128,10 @@ def slice_to_range(
     if stop is None:
         stop = nonempty_domain[1]
 
+    # Indexing beyond end of array should "trim" to end of array
```
Comment (bkmartinjr) [inline apis/python/src/tiledbsoma/util.py:131]: Possibly naive question:  do we need the same for `start` coordinate?   I.e., bound it to zero?

**Review by bkmartinjr — APPROVED** [review]
One question about start range bounding.  Fix for end of range looks great.

File: apis/python/src/tiledbsoma/util.py (hunk 1 above)
Comment (johnkerl) [reply apis/python/src/tiledbsoma/util.py:131]: sure

**Comment by johnkerl** [comment]
@bkmartinjr second thoughts -- we already decided we would not accept negative indices: #519

**Pull request merged**